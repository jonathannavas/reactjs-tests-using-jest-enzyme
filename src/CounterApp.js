import React, { useState } from 'react'
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter( counter + 1 );
    }

    const handleReset = () => {
        setCounter( value );
    }

    const handleSubtract = () => {
        setCounter( counter - 1 );
    }
    
    return (
        <>
            <h4>CounterApp</h4>
            <h5> { counter } </h5>
            <button onClick={ handleAdd } >+1</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSubtract } >-1</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 1
}

export default CounterApp;