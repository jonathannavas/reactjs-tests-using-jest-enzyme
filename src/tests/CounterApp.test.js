import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('Test on CounterApp', () => {
    
    let wrapper = shallow( <CounterApp />);
    
    beforeEach( () => {
        wrapper = shallow( <CounterApp />);
    });

    test('snapshot tith the wrapper', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Show value defect 0', () => {
        const wrapper = shallow( <CounterApp />);
        const value = wrapper.find('h5').text().trim();
        expect( value ).toBe( '0' );
    });

    test('should increment counter bnt+1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h5').text().trim();
        expect( counterText ).toBe('1');
    });

    test('should decrement counter bnt-1', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h5').text().trim();
        expect( counterText ).toBe('-1');
    });
    
    test('should reset counter bnt-reset', () => {
        const wrapper = shallow( <CounterApp value={666} />);
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h5').text().trim();
        expect( counterText ).toBe('666');
    });

});