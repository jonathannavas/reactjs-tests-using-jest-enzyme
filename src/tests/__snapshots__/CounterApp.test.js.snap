// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on CounterApp snapshot tith the wrapper 1`] = `
<Fragment>
  <h4>
    CounterApp
  </h4>
  <h5>
     
    0
     
  </h5>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
